src/render_graph.rs

```diff
@@ -21,22 +40,53 @@ pub fn render_graph_dot(graph: &RenderGraph) -> String {
         let name = node.name.as_deref().unwrap_or("<node>");
         let type_name = utils::short_name(node.type_name);
 
+        let inputs = node
+            .input_slots
+            .iter()
+            .enumerate()
+            .map(|(index, slot)| {
+                format!(
+                    "<TD PORT=\"{}\">{}: {}</TD>",
+                    escape_html(format!("{}", index)),
+                    escape_html(slot.info.name.clone()),
+                    escape_html(format!("{:?}", slot.info.resource_type))
+                )
+            })
+            .collect::<Vec<_>>();
+
         let outputs = node
             .output_slots
             .iter()
-            .map(|slot| format!("{}:{:?}, ", slot.info.name, slot.info.resource_type))
+            .enumerate()
+            .map(|(index, slot)| {
+                format!(
+                    "<TD PORT=\"{}\">{}: {:?}</TD>",
+                    escape_html(format!("{}", index)),
+                    escape_html(slot.info.name.clone()),
+                    escape_html(format!("{:?}", slot.info.resource_type))
+                )
+            })
+            .collect::<Vec<_>>();
+
+        let slots = inputs
+            .iter()
+            .zip_longest(outputs.iter())
+            .map(|pair| match pair {
+                EitherOrBoth::Both(input, output) => format!("<TR>{}{}</TR>", input, output),
+                EitherOrBoth::Left(input) => {
+                    format!("<TR>{}<TD BORDER=\"0\">&nbsp;</TD></TR>", input)
+                }
+                EitherOrBoth::Right(output) => {
+                    format!("<TR><TD BORDER=\"0\">&nbsp;</TD>{}</TR>", output)
+                }
+            })
             .collect::<String>();
-        let outputs = outputs.trim_end_matches(", ");
-        let outputs = match outputs.is_empty() {
-            false => format!("<BR/> {}", font_tag(&format!("-> {}", outputs), "blue", 10)),
-            true => "".into(),
-        };
 
         let label = format!(
-            "<{}<BR />{} {}>",
-            name,
-            font_tag(&type_name, "red", 10),
-            outputs,
+            "<<TABLE><TR><TD PORT=\"title\" BORDER=\"0\" COLSPAN=\"2\">{}<BR/>{}</TD></TR>{}</TABLE>>",
```

Comment:
```suggestion
            "<<TABLE STYLE=\"rounded\"><TR><TD PORT=\"title\" BORDER=\"0\" COLSPAN=\"2\">{}<BR/>{}</TD></TR>{}</TABLE>>",
```
I personally think the rounded edges make the graph more pleasing to look at.

Reply:
Done.